swcc/swcc/models.py

```diff
@@ -372,8 +379,30 @@ def from_name(cls, name: str) -> Optional[Dataset]:
         except StopIteration:
             return None
 
+    def patch_file(self):
+        session = current_session()
+        self.assert_remote()
+        json = self.dict()
+        for key, value in self:
+            if isinstance(value, FileType):
+                json[key] = value.upload()
+        r: requests.Response = session.patch(f'{self._endpoint}/{self.id}/', json=json)
+        raise_for_status(r)
+        cls = self.__class__
+        cache = session.cache[cls]
+        if self.id in cache:
+            del cache[self.id]
+        reload = cls.from_id(cast(int, self.id))
+        self.file = reload.file
+        return self
+
     def load_data_spreadsheet(self, file: Union[Path, str]) -> Dataset:
```

Comment:
```suggestion
    def load_data_spreadsheet(self) -> Dataset:
```

`load_project_spreadsheet` doesn't take any arguments, it just uses the `file` field defined in the constructor. We could require something similar with Datasets:
```
dataset = Dataset(..., file = 'foo/bar/data.xlsx')
dataset.force_create() # uploads the data.xlsx
dataset.load_data_spreadsheet() # throws an exception if file isn't defined, I guess
```

Reply:
I think we'd want to remove the `load_data_spreadsheet` method in this case and have all of that behavior as part of `create`.

If we do that, going forward we can make `file` a required field (but then we have to decide how to migrate old data).